api/envoy/config/trace/v3/xray.proto

```diff
@@ -35,4 +46,9 @@ message XRayConfig {
   // `X-Ray SDK documentation
   // <https://docs.aws.amazon.com/xray/latest/devguide/xray-sdk-go-configuration.html#xray-sdk-go-configuration-sampling>`_
   core.v3.DataSource sampling_rule_manifest = 3;
+
+  // Optional custom fields to be added to each trace segment.
+  // see: `X-Ray Segment Document documentation
+  // <https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html>`__
+  SegmentFields segment_fields = 4;
```

Comment:
qq: should this be repeated? I see that the segment fields has an origin. What if you have different origins? I don't know the protocol but figured I would ask.

Reply:
I though about just making this entire field a struct and pass it through, but I wanted to try and match the x-ray schema 1:1. I also ran this by our X-Ray team and they signed off on this approach.

There is only ever one origin in the sense that it is the origin of the segment, but the entire trace will have multiple origins across its lifetime.

Reply:
:+1: